src/lib/libdylink.js

```diff
@@ -959,11 +980,46 @@ var LibraryDylink = {
      * @param {number=} handle
      * @param {Object=} localScope
      */`,
-  $loadDynamicLibrary: function(libName, flags = {global: true, nodelete: true}, localScope, handle) {
+  $loadDynamicLibrary: function(libName, flags = {global: true, nodelete: true}, localScope, rpath = {parentLibPath: '', paths: []}, handle) {
 #if DYLINK_DEBUG
     dbg(`loadDynamicLibrary: ${libName} handle: ${handle}`);
     dbg(`existing: ${Object.keys(LDSO.loadedLibsByName)}`);
 #endif
+
+#if FILESYSTEM && SUPPORT_RPATH
+      var foundFile = false;
+      if (libName.startsWith("/")) {
+        try {
+          FS.lookupPath(libName);
+          foundFile = true;
+        } catch(e){}
+      } else if (runtimeInitialized) {
+        var runtimePathsAbs = (rpath.paths || []).map((p) => replaceORIGIN(rpath.parentLibPath, p));
```

Comment:
Write this whole block in `withStackSave(() => ..` rather then explicit save and restore?